CalendarPicker/index.js

```diff
@@ -177,31 +177,111 @@ export default class CalendarPicker extends Component {
       onDateChange,
       enableDateChange,
       allowBackwardRangeSelect,
+      maxRangeDuration,
     } = this.props;
 
     if (!enableDateChange) {
       return;
     }
 
-    const date = moment({ year: currentYear, month: currentMonth, day, hour: 12 });
+    const date = moment({ year: currentYear, month: currentMonth, day, hour: 12 }),
+      startDate = selectedStartDate
+        ? moment(selectedStartDate)
+        : selectedStartDate,
+      endDate = selectedEndDate ? moment(selectedEndDate) : selectedEndDate;
```

Comment:
This could be simplified to:
```
startDate = selectedStartDate && moment(selectedStartDate);
endDate = selectedEndDate && moment(selectedEndDate);
```
If the date is undefined, that returns undefined.  If it's an object, it will create a moment object.  That should have the same result as your ternary.

Reply:
updated in https://github.com/stephy/CalendarPicker/pull/196